src/test/java/uk/gov/hmcts/reform/cwrdapi/TestSupport.java

```diff
@@ -86,6 +94,144 @@ public static List<CaseWorkerDomain> buildSuspendedCaseWorkerProfileData() {
                 .build());
     }
 
+
+    public static boolean validateSearchUserProfileResponse(ResponseEntity<List<SearchStaffUserResponse>> response,
+                                                            SearchRequest searchReq) {
+
+        SearchRequest finalSearchReq = searchReq;
+        List validResponse = new ArrayList<>();
+        List<SearchStaffUserResponse> body = response.getBody();
+        validResponse = body.stream()
+                .filter(Objects::nonNull)
+                .filter(searchStaffUserResponse -> {
+                    if (Optional.ofNullable(finalSearchReq.getServiceCode()).isEmpty()) {
+                        return true;
+                    }
+                    if (Optional.ofNullable(searchStaffUserResponse.getServices()).isEmpty()) {
+                        return false;
+                    }
+
+                    boolean valid = searchStaffUserResponse.getServices().stream()
+                            .anyMatch(service -> Optional.ofNullable(service).isPresent()
+                                    && finalSearchReq.getServiceCode().toLowerCase().contains(service.getServiceCode()
+                                    .toLowerCase())
+                            );
+                    return valid;
+                })
+                .filter(searchStaffUserResponse -> {
+                    if (Optional.ofNullable(finalSearchReq.getLocation()).isEmpty()) {
+                        return true;
+                    }
+                    if (Optional.ofNullable(searchStaffUserResponse.getBaseLocations()).isEmpty()) {
+                        return false;
+                    }
+
+                    boolean valid = searchStaffUserResponse.getBaseLocations().stream()
+
+                            .anyMatch(location -> Optional.ofNullable(location).isPresent()
+                                    && finalSearchReq.getLocation().toLowerCase().contains(location.getBaseLocationId()
+                                    .toString().toLowerCase())
+                            );
+                    return valid;
+                })
+                .filter(searchStaffUserResponse -> {
+                    if (Optional.ofNullable(finalSearchReq.getLocation()).isEmpty()) {
```

Comment:
duplicated , its already covered in above filter

Reply:
changes done